app/scripts/utils/track-utils.js

```diff
@@ -0,0 +1,404 @@
+
+    const tiles = xTiles.map(x => [zoomLevel, x]);
+
+    return tiles;
+  }
+
+  // x doesn't necessary mean 'x' axis, it just refers to the relevant axis
+  // (x if horizontal, y if vertical)
+  const xTiles = calculateTiles(
+    zoomLevel,
+    scale,
+    tilesetInfo.min_pos[0],
+    tilesetInfo.max_pos[0],
+    tilesetInfo.max_zoom,
+    tilesetInfo.max_width
+  );
+
+  const tiles = xTiles.map(x => [zoomLevel, x]);
+  return tiles;
+};
+
+/**
+ * Draw an axis on a track. Where on the track the axis will be drawn
+ * is taken from the track's options.
+ *
+ * @param  {PixiTrack} track   The track to decorate with an axis.
+ * @param  {d3.scale} valueScale The scale that the axis should draw.
+ */
+const drawAxis = (track, valueScale) => {
+  if (!track.axis) {
+    track.axis = new AxisPixi(track);
+    track.pBase.addChild(track.axis.pAxis);
+  }
+  // either no axis position is specified
+  if (!track.options.axisPositionVertical && !track.options.axisPositionHorizontal) {
+    track.axis.clearAxis();
+    return;
+  }
+
+  if (track.options.axisPositionVertical && track.options.axisPositionVertical === 'hidden') {
+    track.axis.clearAxis();
+    return;
+  }
+
+  if (track.options.axisPositionHorizontal && track.options.axisPositionHorizontal === 'hidden') {
+    track.axis.clearAxis();
+    return;
+  }
+
+  const margin = track.options.axisMargin || 0;
+
+  if (
+    track.options.axisPositionHorizontal === 'left'
+    || track.options.axisPositionVertical === 'top'
+  ) {
+    // left axis are shown at the beginning of the plot
+    track.axis.pAxis.position.x = track.position[0] + margin;
+    track.axis.pAxis.position.y = track.position[1];
+
+    track.axis.drawAxisRight(valueScale, track.dimensions[1]);
+  } else if (
+    track.options.axisPositionHorizontal === 'outsideLeft'
+    || track.options.axisPositionVertical === 'outsideTop'
+  ) {
+    // left axis are shown at the beginning of the plot
+    track.axis.pAxis.position.x = track.position[0] + margin;
+    track.axis.pAxis.position.y = track.position[1];
+
+    track.axis.drawAxisLeft(valueScale, track.dimensions[1]);
+  } else if (
+    track.options.axisPositionHorizontal === 'right'
+    || track.options.axisPositionVertical === 'bottom'
+  ) {
+    track.axis.pAxis.position.x = track.position[0] + track.dimensions[0] - margin;
+    track.axis.pAxis.position.y = track.position[1];
+    track.axis.drawAxisLeft(valueScale, track.dimensions[1]);
+  } else if (
+    track.options.axisPositionHorizontal === 'outsideRight'
+    || track.options.axisPositionVertical === 'outsideBottom'
+  ) {
+    track.axis.pAxis.position.x = track.position[0] + track.dimensions[0] - margin;
+    track.axis.pAxis.position.y = track.position[1];
+    track.axis.drawAxisRight(valueScale, track.dimensions[1]);
+  }
+};
+
+/**
+ * A track is being dragged along it's value scale axis.
+ *
+ * @param {PixiTrack} Track The track being dragged.
+ * @param  {Number} dY The change in y position.
```

Comment:
```suggestion
 * @param  {number} dY The change in y position.
```

Just for consistency :)